mm/kasan/kasan.c

```diff
@@ -107,12 +107,12 @@ static void kasan_report(uintptr_t addr, size_t size, bool is_write)
   --recursion;
 }
 
-static bool kasan_is_poisoned(uintptr_t addr, size_t size)
+static bool kasan_is_poisoned(FAR void *addr, size_t size)
 {
   FAR uintptr_t *p;
   unsigned int bit;
 
-  p = kasan_mem_to_shadow(addr + size - 1, 1, &bit);
+  p = kasan_mem_to_shadow((uintptr_t)(addr + size - 1), 1, &bit);
```

Comment:
let's change kasan_mem_to_shadow and kasan_set_poison prototype to FAR void * too